command_list.md

```diff
@@ -0,0 +1,132 @@
+# Account
+```
+account [list]
+account add 
+account info 
+account remove 
+account usage 
+account set
+
+account ban add
+account ban remove
+
+account attribute list 
+account attribute add 
+account attribute remove 
+
+account limit list 
+account limit add 
+account limit remove
+
+account identity list
+account identity add 
+account identity remove 
+```
+
+# Config 
+```
+config [list]
+config set
+config remove
+```
+
+# DID 
+```
+did [list]
+did stat
+did remove
+did touch
+did parent
+did attach
+did detach
+did upload
+did download
+
+did content list
+did content history
+did content close
+did content open
+
+did metadata list
+did metadata add
+did metadata remove
+
+did container add
+
+did dataset add
+```
+
+# Replica 
+```
+replica [list]
+replica dataset 
+replica pfn 
+replica tombstone
+
+replica state suspicious
+replica state quarantine
+replica state bad
+replica state temp-unavailable
+```
+
+# RSE
+```
+rse [list]
+rse info
+rse remove
+rse add
+rse set
+
+rse attribute list
+rse attribute set
+rse attribute remove
+
+rse distance list
+rse distance add
+rse distance remove
+rse distance set
+
+rse protocol add
+rse protocol remove
+
+rse limit add
+rse limit remove
+
+rse qos list
+rse qos add
+rse qos remove
+
+```
+
+# Rule 
+```
+rule  add
+rule  remove
+rule  info
+rule  history
+rule  set
+rule  [list]
```

Comment:
This helps a lot. I wish to start with the commands related to replication rules; it seems the easiest to start a discussion.

* `add` is fine. I thought about `create`, which is more consistent with the `created_at` columns in the database, but then maybe it doesn’t fit with some other subjects (e.g. ‘creating’ an RSE distance).
* `remove` is similar. I thought about `delete` (consistent with `deleted_at`), but ‘remove’ is a better antonym to ‘add’.
* `info` I would suggest to rename to `show`. I find it preferable because it is a verb.
* `history` is completely fine.
* `set` I have some trouble with. It creates the expectation that `unset` exists as well. I think `update` is a better choice. 
* `list` is perfect.

Please note that this is my personal opinion. I’m waiting for you and others to offer theirs.

Reply:
>  I thought about create, which is more consistent with the created_at columns in the database, but then maybe it doesn’t fit with some other subjects (e.g. ‘creating’ an RSE distance). 

I prefer add/remove for this exact reason. Also might just be because the main cli tool I use is git so add/remove is nice and familiar to me, personally. 

> info I would suggest to rename to show. I find it preferable because it is a verb. 

`info` was picked because it's the same verbiage (noun-iage?) from the old client. But at this point, there is no need to prioritize familiarity, so I think `show` is the better option. There's a few other instances of `info` that would be impacted by this. 

> set I have some trouble with. It creates the expectation that unset exists as well. I think update is a better choice. 

Another carry-over from the old client. The original proposal also included an `unset` as it's antonym, so it made more sense there. There isn't a single `unset` in this version so `update` should be used.

Reply:
I concur with your comments. I am happy that we’re on the same page.

Then let me offer a quick summary:

1. Commands are grouped around *entities* (e.g. a replication rule).
2. There are subcommands to `list` existing entities, `show` more information about a specific entity, `add` a new entity, `update` the information of a specific entity, and `remove` a specific entity.

So, I believe that going forward:

1. We should try to offer this structure for every possible entity.
2. Some entities may require special subcommands (e.g. `history`). We should not be afraid to add them.
3. Some commands from the current client may not fit with the new structure and may be considered unnecessary. We should not be afraid to remove them.
4. We might want to consider whether some subcommands need to be ‘flexible’ (e.g. whether `add`ing an RSE attribute that already exists should automatically update it and, likewise, whether `update`ing a non-existent RSE attribute should automatically add it). I prefer being explicit, but this feels like a quality-of-life feature. The logging should be clear when this happens.